ons_alpha/articles/models.py

```diff
@@ -0,0 +1,187 @@
+
+    content_panels = (
+        BasePage.content_panels
+        + BundledPageMixin.panels
+        + [
+            FieldPanel("summary"),
+            MultiFieldPanel(
+                [
+                    FieldRowPanel(
+                        [
+                            FieldPanel("release_date"),
+                            FieldPanel("next_release_date"),
+                        ]
+                    ),
+                    FieldPanel("is_accredited"),
+                    FieldPanel("contact_details"),
+                ],
+                heading="Metadata",
+            ),
+            FieldPanel("body"),
+        ]
+    )
+
+    edit_handler = TabbedInterface(
+        [
+            ObjectList(content_panels, heading="Content"),
+            ObjectList(
+                [
+                    MultipleChooserPanel("topics", label="Topic", chooser_field_name="topic"),
+                ],
+                help_text="Select the topics that this article relates to.",
+                heading="Taxonomy",
+            ),
+            ObjectList(
+                [FieldPanel("updates", help_text="Add any corrections or updates")], heading="Corrections & Updates"
+            ),
+            ObjectList(BasePage.promote_panels, heading="Promote"),
+            ObjectList(BasePage.settings_panels, heading="Settings", classname="settings"),
+        ]
+    )
+
+    search_fields = BasePage.search_fields + [
+        index.SearchField("summary"),
+        index.SearchField("body"),
+        index.SearchField("get_admin_display_title", boost=2),
+        index.AutocompleteField("get_admin_display_title"),
+    ]
+
+    @property
+    def full_title(self):
+        return f"{self.get_parent().title}: {self.title}"
+
+    @property
+    def document_type(self):
+        return _("Article")
+
+    @property
+    def is_latest(self):
+        latest_id = (
+            ArticlePage.objects.sibling_of(self).live().order_by("-release_date").values_list("id", flat=True).first()
+        )
+        return self.pk == latest_id
+
+    @cached_property
+    def toc(self):
+        items = [{"url": "#summary", "text": "Summary"}]
+        for block in self.body:  # pylint: disable=not-an-iterable
+            if hasattr(block.block, "to_table_of_contents_items"):
+                items += block.block.to_table_of_contents_items(block.value)
+        if self.contact_details_id:
+            items += [{"url": "#contact-details", "text": "Contact details"}]
+        return items
+
+    def get_admin_display_title(self):
+        # tweak the admin display title to include the parent title
+        return f"{self.get_parent().title}: {self.draft_title or self.title}"
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+        context["toc"] = self.toc
+        return context
+
+    @path("previous/v<int:version>/")
+    def previous_version(self, request, version):
```

Comment:
Suggestion: Much of this updates code is duplicated from bulletins. Perhaps it should live in a mixin?

Reply:
the PR description explicitly calls this out.